system/BaseModel.php

```diff
@@ -899,6 +899,10 @@ public function insertBatch(?array $set = null, ?bool $escape = null, int $batch
      */
     public function update($id = null, $data = null): bool
     {
+        if (is_bool($id)) {
```

Comment:
In what instances would this be a bool?

Reply:
When an attacker passes a bool and the dev forgets to validate the id value.